### PR TITLE
For now only allow 1 link share per file/folder

### LINKED ISSUE #22692: A user can create an unlimited amount of duplicate public links for a single folder when using the API
### Steps to reproduce
1. Create a new folder share using the API
2. Create it again using the same POST data

URL: `/ocs/v2.php/apps/files_sharing/api/v1/shares?format=json`
Payload: `path=pics%2Fbrand+guideline+logos&shareType=3&shareWith=&publicUpload=0&password=&permissions=1`

### Expected behaviour

The second request should not go through and a specific error should be returned, indicating to the client that a link already exists

OR

The link which was created earlier should be returned

### Actual behaviour

Each request returns a new link object, with a different ID and token.

### Server configuration

**ownCloud version:** master

#### Browser log

1st request

```
id: 624, share_type: 3, uid_owner: "admin", displayname_owner: "admin", permissions: 1, stime: 1456584648, parent: null, expiration: null, token: "uaidgEJbXb3DQzj", uid_file_owner: "admin",...
```

2nd request

```
id: 625, share_type: 3, uid_owner: "admin", displayname_owner: "admin", permissions: 1, stime: 1456584745, parent: null, expiration: null, token: "V1azHym3B4EchAf", uid_file_owner: "admin",...
```

----
@rullzer @PVince81 @MorrisJobke

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #22692
This is a temp fix util we get #22327

We just block it in the OCS Share API since we have 1 endpoint now anyway.

### CODE REVIEW

File: apps/files_sharing/api/share20ocs.php [hunk 1]
```diff
@@ -303,6 +303,15 @@ public function createShare() {
 				return new \OC_OCS_Result(null, 404, 'public link sharing is disabled by the administrator');
 			}
 
+			/*
+			 * For now we only allow 1 link share.
+			 * Return the existing link share if this is a duplicate
+			 */
+			$existingShares = $this->shareManager->getSharesBy($this->currentUser->getUID(), \OCP\Share::SHARE_TYPE_LINK, $path, false, 1, 0);
+			if (count($existingShares) === 1) {
```
Comment (nickvergessen) [inline apps/files_sharing/api/share20ocs.php]: Id check `if (!empty(`, otherwise if someone already has two, he can create more

Comment (rullzer) [inline apps/files_sharing/api/share20ocs.php]: good point. fixed.

**Comment by nickvergessen** [comment]
:+1: 

@rullzer missing the "To Review" label?

**Comment by rullzer** [comment]
@nickvergessen yeah ;)

@PVince81 @cmonteroluque your call... 9.0.0 or 9.0.1

**Comment by PVince81** [comment]
I don't think this is critical, the API always behaved this way in the past.

Moving to 9.0.1.

CC @cmonteroluque

**Comment by rullzer** [comment]
@PVince81 actually no. In 8.2.x you could only create 1 link share. In 9.0.0 you can now create more.

**Comment by PVince81** [comment]
Can we have an integration test for this ?

CC @SergioBertolinSG

**Comment by PVince81** [comment]
This is indeed a bug / change of behavior and might affect clients that rely on the old behavior.
Moving to 9.0.0

**Comment by rullzer** [comment]
I'll add the intergration tests

**Comment by PVince81** [comment]
Tested, works :+1:

**Comment by karlitschek** [comment]
:+1: looks good.

**Comment by DeepDiver1975** [comment]
requires rebase

**Comment by DeepDiver1975** [comment]
rebased

**Pull request merged**